src/nitrokey/trussed/_bootloader/nrf52.py

```diff
@@ -45,18 +46,23 @@ class SignatureKey:
     # $ openssl ec -in dfu_public.pem -inform pem -pubin -outform der | xxd -p
     der: str
 
-    def vk(self) -> ecdsa.VerifyingKey:
-        return ecdsa.VerifyingKey.from_der(bytes.fromhex(self.der))
+    def public_key(self) -> ec.EllipticCurvePublicKey:
+        public_key = serialization.load_der_public_key(bytes.fromhex(self.der))
+        assert isinstance(public_key, ec.EllipticCurvePublicKey)
+        return public_key
```

Comment:
Shouldn't this also check the actual algorithm used?

Reply:
You mean the curve?  Sure, I can do that, but I don’t see the need.